src/Couchbase.Lite.Shared/API/Database/Database.cs

```diff
@@ -804,8 +786,17 @@ public bool SetDocumentExpiration(string docId, DateTimeOffset? timestamp)
             if (timestamp == null) {
                 return Native.c4doc_setExpiration(_c4db, docId, 0, null);
             }
-            var Timestamp = timestamp?.ToUnixTimeMilliseconds();
-            return Native.c4doc_setExpiration(_c4db, docId, (ulong)Timestamp, null);
+            var succeed = false;
+            ThreadSafety.DoLocked(() =>
+            {
+                LiteCoreBridge.Check(err =>
+                {
```

Comment:
This can be combined with the convenience function `ThreadSafety.DoLockedBridge`